tests/cli/test_path_validation.py

```
@@ -0,0 +1,229 @@
+        assert str(missing.resolve()) in message
+
+    def test_resolution_failure_becomes_bad_parameter(
+        self, tmp_path: Path, monkeypatch: pytest.MonkeyPatch
+    ) -> None:
+        """``Path.resolve`` can raise ``RuntimeError`` (py<3.13) or ``OSError``
+        (py>=3.13) on symlink loops and unresolvable ``~user``. The helper's
+        contract is to surface these as ``typer.BadParameter`` so typer emits
+        a usage error (exit 2) instead of an internal traceback.
+        """
+
+        def _boom(self: Path, *_: object, **__: object) -> Path:
```

Ruff (0.15.10)

[warning] Unused function argument: `self`

(ARG001)